### ISSUE TITLE
Add MLPRegressor from scikit learn

### ISSUE THREAD

**Issue by gsheni** [alteryx/evalml#3537]
- There is a regressor we could add to EvalML:
  - https://scikit-learn.org/stable/modules/generated/sklearn.neural_network.MLPRegressor.html
- This came up because we want to properly treat cyclic categoricals
  - http://blog.davidkaleko.com/feature-engineering-cyclical-features.html
  - https://towardsdatascience.com/how-to-handle-cyclical-data-in-machine-learning-3e0336f7f97c
- Cyclic categorical features won't matter for decision tree based algorithms:
>  Decision trees based algorithms (Random Forest, Gradient Boosted Trees, XGBoost) build their split rules according to one feature at a time. This means that they will fail to process these two features simultaneously whereas the cos/sin values are expected to be considered as one single coordinates system.
